nautobot/docs/apps/migrating-jobs-from-nautobot-v1.md

```diff
@@ -0,0 +1,157 @@
+            return data["a"] + data["b"]
+    ```
+
+    ```py title="v2.0 Job"
+    class AddJob(Job):
+        a = IntegerVar()
+        b = IntegerVar()
+
+        def run(self, a, b):
+            return a + b
+    ```
+
+### `test_*` and `post_run()` Methods
+
+The `test_*` and `post_run` methods for backwards compatibility to NetBox scripts and reports were removed. Celery implements `before_start`, `on_success`, `on_retry`, `on_failure`, and `after_return` methods that can be used by job authors to perform similar functions.
+
+!!! important
+    Be sure to call the `super()` method when overloading any of the job's `before_start`, `on_success`, `on_retry`, `on_failure`, or `after_return` methods
+
+### Database Transaction Handling
+
+Jobs no longer run in an atomic database transaction by default. If a Job needs to run in a database transaction, you can use the `@transaction.atomic` decorator on the `run()` method or wrap parts of your Job code in the `with transaction.atomic()` context manager.
+
+!!! example
+    ```py
+    from django.db import transaction
+    from nautobot.dcim import models
+    from nautobot.extras.jobs import Job, ObjectVar
+
+    class UpdateDeviceTypeHeightJob(Job):
+        device_type = ObjectVar(model=models.DeviceType)
+        u_height = IntegerVar(label="New Height (U)")
+
+        @transaction.atomic
+        def run(self, device_type, u_height):
+            device_type.u_height = u_height
+            device_type.save()
+    ```
+
+#### Commit Argument
+
+As a result of the default database transaction being removed from Nautobot core, the `commit` argument has been removed. If a Job author wants to provide users the ability to bypass approval when `approval_required` is set, the Job must implement a `dryrun` variable using the newly introduced `DryRunVar`. The desired value of the variable will be passed to the run method just like any other variable but the Job author must implement the logic to handle the dryrun.
+
+The presence of a `dryrun = DryRunVar()` property on the Job class sets the `supports_dryrun` flag on the Job model, which allows users to bypass approval when `approval_required` is set. To implement a dryrun variable without allowing users to bypass approval, the `dryrun` variable should use the `BooleanVar` class instead of `DryRunVar`.
+
+The Job `commit_default` property has been renamed to `dryrun_default` and the default value as well as any existing database values have been flipped.
+
+#### Read-Only Meta Attribute
+
+The `read_only` Job field no longer changes the behavior of Nautobot core and is informational only in v2.0.
+
+### Job Registration
+
+All Jobs must be registered in the celery task registry to be available in Nautobot. This must be accomplished by calling `nautobot.core.celery.register_jobs(*job_classes)` at the top level of a Job module so that it is registered when the module is imported. The `register_jobs` method accepts one or more job classes as arguments.
```

Comment:
```suggestion
All Jobs must be registered in the Celery task registry to be available in Nautobot. This must be accomplished by calling `nautobot.core.celery.register_jobs(*job_classes)` at the top level of a Job module so that it is registered when the module is imported. The `register_jobs` method accepts one or more job classes as arguments.
```